### ISSUE TITLE
Document when to call markNeedsSemanticsUpdate

### ISSUE THREAD

**Issue by goderbauer** [flutter/flutter#11716]
On the `SemanticsNode` class it should be documented when `markNeedsSemanticsUpdate` needs to be called (with regards to changing the flags, actions, and tags of a `SemanticsNode`).

Also, the documentation of the parameters (and when to set them to true/false) of `markNeedsSemanticsUpdate` should be improved.

**Comment by goderbauer** [comment]
Fixed in https://github.com/flutter/flutter/pull/11770.
